packages/cli/src/executions/active-execution.service.ts

```diff
@@ -0,0 +1,143 @@
+		private readonly activeExecutions: ActiveExecutions,
+		private readonly executionRepository: ExecutionRepository,
+		private readonly waitTracker: WaitTracker,
+	) {}
+
+	async findOne(executionId: string, accessibleWorkflowIds: string[]) {
+		return await this.executionRepository.findIfAccessible(executionId, accessibleWorkflowIds);
+	}
+
+	private toSummary(execution: IExecutionsCurrentSummary | IExecutionBase): ExecutionSummary {
+		return {
+			id: execution.id,
+			workflowId: execution.workflowId ?? '',
+			mode: execution.mode,
+			retryOf: execution.retryOf !== null ? execution.retryOf : undefined,
+			startedAt: new Date(execution.startedAt),
+			status: execution.status,
+			stoppedAt: 'stoppedAt' in execution ? execution.stoppedAt : undefined,
+		};
+	}
+
+	// ----------------------------------
+	//           regular mode
+	// ----------------------------------
+
+	async findManyInRegularMode(
+		filter: GetManyActiveFilter,
+		accessibleWorkflowIds: string[],
+	): Promise<ExecutionSummary[]> {
+		return this.activeExecutions
+			.getActiveExecutions()
+			.filter(({ workflowId }) => {
+				if (filter.workflowId && filter.workflowId !== workflowId) return false;
+				if (workflowId && !accessibleWorkflowIds.includes(workflowId)) return false;
+				return true;
+			})
+			.map((execution) => this.toSummary(execution))
+			.sort((a, b) => Number(b.id) - Number(a.id));
+	}
+
+	async stopOneInRegularMode(execution: IExecutionBase) {
+		const result = await this.activeExecutions.stopExecution(execution.id);
+
+		if (result) {
+			return {
+				mode: result.mode,
+				startedAt: new Date(result.startedAt),
+				stoppedAt: result.stoppedAt ? new Date(result.stoppedAt) : undefined,
+				finished: result.finished,
+				status: result.status,
+			};
+		}
+
+		return await this.waitTracker.stopExecution(execution.id);
+	}
+
+	// ----------------------------------
+	//           queue mode
+	// ----------------------------------
+
+	async findManyInQueueMode(filter: GetManyActiveFilter, accessibleWorkflowIds: string[]) {
+		const activeManualExecutionIds = this.activeExecutions
+			.getActiveExecutions()
+			.map((execution) => execution.id);
+
+		const activeJobs = await this.queue.getJobs(['active', 'waiting']);
+
+		const activeProductionExecutionIds = activeJobs.map((job) => job.data.executionId);
+
+		const activeExecutionIds = activeProductionExecutionIds.concat(activeManualExecutionIds);
+
+		if (activeExecutionIds.length === 0) return [];
+
+		const activeExecutions = await this.executionRepository.getManyActive(
+			activeExecutionIds,
+			accessibleWorkflowIds,
+			filter,
+		);
+
+		return activeExecutions.map((execution) => {
+			if (!execution.status) {
+				// @tech-debt Status should never be nullish
+				execution.status = getStatusUsingPreviousExecutionStatusMethod(execution);
+			}
+
+			return this.toSummary(execution);
+		});
+	}
+
+	async stopOneInQueueMode(execution: IExecutionBase): Promise<IExecutionsStopData> {
```

Comment:
There's some duplication in the two methods, maybe it makes sense to call both `stopExecution` and if in queue mode, continue with the queue specific code wdyt?